sdk/src/azure/iot/az_iot_hub_client_methods.c

```diff
@@ -39,7 +39,10 @@ AZ_NODISCARD az_result az_iot_hub_client_methods_parse_received_topic(
     return AZ_ERROR_IOT_TOPIC_NO_MATCH;
   }
 
-  _az_LOG_WRITE(AZ_LOG_MQTT_RECEIVED_TOPIC, received_topic);
+  if (_az_LOG_SHOULD_WRITE(AZ_LOG_MQTT_RECEIVED_TOPIC))
```

Comment:
Can we add some tests please? See existing tests like:
https://github.com/Azure/azure-sdk-for-c/blob/07c055c11b6bac12b8e547f92ee181204095fa1e/sdk/tests/iot/common/test_az_iot_common.c#L247-L259

Reply:
added negative tests for all places in IoT which have a positive test